### PR TITLE
Introduction module complete

### PR DESCRIPTION
**Pull request by SiarheiMatsevich**

### CODE REVIEW

File: 1-introduction/introduction_Task1.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+for i in range(1, 101):
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task1.py]: There are no questions to the implementation, everything is good, but I would ask to formalize all tasks as a separate function with docstrings and description of variables, and the main part. Like following

```python
def function_name(var_name: var_type) -> function_tupe:
    """Function description.

    Vars description
    """
    pass


if __name__ == "__main__":
    function_name(var_name=var_value)
```

This is good code writing practice, so get used to writing like this right away. This comment also applies to all tasks, including future ones.

File: 1-introduction/introduction_Task2.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+numbers = [1, 2, '0', '300', -2.5, 'Dog', True, 0o1256, None]
+for n, i in enumerate(numbers):
+    try:
+        numbers[n] = int(i)
+    except TypeError:
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task2.py]: 1. You can use

```python
except (TypeError, ValueError):
```

2. You will fail if input list is empty
3. It is bad practice to change the original variable if it is not necessary. Create a new list and add new values to it.

File: 1-introduction/introduction_Task3.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+implementations. Python and CPython are managed by the non-profit Python Software Foundation.'''
+
+normalized_str = input_str.lower()
+count = [normalized_str.count(i) for i in string.ascii_lowercase]
+top_used = string.ascii_lowercase[count.index(max(count))]
+
+# If the word 'Python' should be also case inventive than we should use 'normalized_str' variable instead of 'input_str'
+# In this particular case it does not matter since the word 'Python' is always typed with leading capital latter.
+print(f'Letter "{top_used}" is used most often')
+print(f'The word "Python" is used {input_str.count('Python')} times.')
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task3.py]: What version of python did you use? I get an error for version 3.11.
```
    print(f'The word "Python" is used {input_str.count('Python')} times.')
                                                        ^^^^^^
SyntaxError: f-string: unmatched '('
```

File: 1-introduction/introduction_Task4.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from sys import maxsize
+
+
+def file_size(size_in_bytes):
+    def converter(value, unit='B'):
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task4.py:10]: Good solution! Although in my opinion it is redundant here and complicates code readability. I'll leave it up to you, but if you want, make it simpler and use a regular loop.

File: 1-introduction/introduction_Task5.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+def get_sum(seq, n):
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task5.py]: Cosmetis: The name of the function does not reflect what it does in my opinion.

File: 1-introduction/introduction_Task5.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+    return n * ((seq - 1) // n) * ((seq - 1) // n + 1) // 2
+
+
+l = 100000000
+print(get_sum(l, 3) + get_sum(l, 5) - get_sum(l, 15))
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task5.py]: I would make a separate function for this

File: 1-introduction/standard_lib_Task1.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+l = [1, 2, '3', 4, None, 10, 33, 'Python', -37.5]
+
+
+def filter_list(l):
+
+    # •	“for” loop
+    def for_filter_list():
+        for i in l:
+            if type(i) is not int:
+                l.remove(i)
```
Comment (akhmialeuski) [inline 1-introduction/standard_lib_Task1.py]: You should not use remove and modify the original list

File: 1-introduction/standard_lib_Task1.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+    # •	filter() + lambda
+    def lambda_filter():
+        return list(filter(lambda i: type(i) is int, l))
+
+    functions = random.choice((for_filter_list, list_compr, lambda_filter))
+    return functions()
+
+
+filter_list([1,2,'a','b']) == [1,2]
```
Comment (akhmialeuski) [inline 1-introduction/standard_lib_Task1.py]: I'd use `assert` here

File: 1-introduction/standard_lib_Task2.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
+
+def check_ip(ip_address):
```
Comment (akhmialeuski) [inline 1-introduction/standard_lib_Task2.py]: good option

File: 1-introduction/standard_lib_Task3.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+from csv import DictReader
+from json import dump
+
+
+with open('cars.csv') as f:
+    reader = DictReader(f)
+    csv_dict = [row for row in reader]
```
Comment (akhmialeuski) [inline 1-introduction/standard_lib_Task3.py]: I'd make it simple like `csv_dict = list(reader)`

**Review by akhmialeuski — CHANGES_REQUESTED** [review]

File: 1-introduction/introduction_Task1.py (hunk 1 above)
Comment (SiarheiMatsevich) [reply 1-introduction/introduction_Task1.py]: Fixed. Thanks!

File: 1-introduction/introduction_Task2.py (hunk 2 above)
Comment (SiarheiMatsevich) [reply 1-introduction/introduction_Task2.py]: I've refactored the whole logic - the try/except block has been removed
Please review

File: 1-introduction/introduction_Task3.py (hunk 3 above)
Comment (SiarheiMatsevich) [reply 1-introduction/introduction_Task3.py]: I use Python 3.12 but I updated the function, so now it should work fine with older version as well

File: 1-introduction/introduction_Task5.py (hunk 5 above)
Comment (SiarheiMatsevich) [reply 1-introduction/introduction_Task5.py]: Fixed

File: 1-introduction/introduction_Task5.py (hunk 6 above)
Comment (SiarheiMatsevich) [reply 1-introduction/introduction_Task5.py]: I've changed initial implementation - please review

File: 1-introduction/standard_lib_Task1.py (hunk 7 above)
Comment (SiarheiMatsevich) [reply 1-introduction/standard_lib_Task1.py]: This is fixed - thanks!

File: 1-introduction/standard_lib_Task1.py (hunk 8 above)
Comment (SiarheiMatsevich) [reply 1-introduction/standard_lib_Task1.py]: Added - thanks!

File: 1-introduction/standard_lib_Task3.py (hunk 10 above)
Comment (SiarheiMatsevich) [reply 1-introduction/standard_lib_Task3.py]: That's a good point - I've updated this line - thank you!

File: 1-introduction/introduction_Task3.py [hunk 11]
```diff
@@ -10,11 +11,20 @@
...
+    normalized_str = text.lower()
+    count = [normalized_str.count(i) for i in string.ascii_lowercase]
+    top_used = string.ascii_lowercase[count.index(max(count))]
+    word_count = normalized_str.count(word)
+    print(f'Letter "{top_used}" is used most often')
+    print(f'The word "Python" is used {word_count} times.')
+
+
+if __name__ == "__main__":
+    get_most_used_letter_and_word_count(input_str, 'Python')
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task3.py]: It's not working right.
```
Letter "e" is used most often
The word "Python" is used 0 times.
```

File: 1-introduction/introduction_Task2.py [hunk 12]
```diff
@@ -1,11 +1,23 @@
...
-    except TypeError:
-        numbers.pop(n)
-    except ValueError:
-        numbers.pop(n)
-
-print(f'Maximum value is: {max(numbers)}')
-print(f'Minimum value is: {min(numbers)}')
+
+
+def get_min_max(li: List) -> None:
```
Comment (akhmialeuski) [inline 1-introduction/introduction_Task2.py]: It is good practice not to print the result of a function in the function itself, but to output the result and then print it to the screen (except in rare cases or for debugging).

Comment (akhmialeuski) [reply 1-introduction/introduction_Task2.py]: It will be problematic to test it if unit tests are created

Comment (akhmialeuski) [reply 1-introduction/introduction_Task2.py]: Same for a few other task.

**Review by akhmialeuski — APPROVED** [review]
well done

**Pull request merged**